extractor/src/main/java/org/schabi/newpipe/extractor/channel/ChannelInfo.java

```diff
@@ -57,9 +57,31 @@ public static ChannelInfo getInfo(StreamingService service, String url) throws I
 
     public static ChannelInfo getInfo(ChannelExtractor extractor) throws IOException, ExtractionException {
 
-        ChannelInfo info = new ChannelInfo(extractor.getServiceId(),
-                extractor.getLinkHandler(),
-                extractor.getName());
+        ChannelInfo info;
+        try {
+            info = new ChannelInfo(extractor.getServiceId(),
+                    extractor.getLinkHandler(),
+                    extractor.getName());
+        } catch (ParsingException e) {
+            info = new ChannelInfo(extractor.getServiceId(),
+                    extractor.getLinkHandler(),
+                    "");
+            Throwable nameFailure;
+            if (extractor instanceof YoutubeChannelExtractor) {
```

Comment:
the abstract class "Info" could have the new method "public abstract String obtainErrorMessage". It wouldn't do anything yet for other extractors but the YoutubeExtractor, but this would be an approach that wouldn't require casting to the YoutubeChannelExtractor then.

Reply:
Yeah, you could add `obtainErrorMessage` to the base `Info` class and make it return `null` by default. When extractors override it, they can return a `String` to indicate there was an error or return null otherwise (as you did with youtube).

Reply:
sorry, Info is not the correct inheritance hierarchy. I have made a new commit where I have added the method obtainErrorMessage in the abstract class ChannelExtractor, returning null by default, as you have suggested @Stypox